docker_entrypoint.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+addgroup -g ${GID} mastodon && adduser -h /mastodon -s /bin/sh -D -G mastodon -u ${UID} mastodon
+find /mastodon -path /mastodon/public/system -prune -o -not -user mastodon -not -group mastodon -print0 | xargs -0 chown -f mastodon:mastodon
```

Comment:
This might have a huge performance impact and is pretty much in contradiction with Docker's philosophy of "start fast". We can already see the effects: https://github.com/tootsuite/mastodon/issues/3194

Seems like a bad tradeoff only to be able to set UID/GID.

If someone really wants to use a specific UID/GID, maybe use build args with defaults instead. So they can rebuild the image with whatever UID/GID they want without hampering the container boot time.

Reply:
> This might have a huge performance impact and is pretty much in contradiction with Docker's philosophy of "start fast". We can already see the effects: #3194
> 
> Seems like a bad tradeoff only to be able to set UID/GID.
> 
> If someone really wants to use a specific UID/GID, maybe use build args with defaults instead. So they can rebuild the image with whatever UID/GID they want without hampering the container boot time.

e

Reply:
@Wonderfall I just had this idea: How about checking permissions on *one* file, and only running the update if they are wrong? There's like 0 chance that permissions will be mixed, so if one file's good, the others are too. This will make startup as fast as before in most cases.